packages/react-notion-x/src/third-party/collection-view-calendar.tsx

```diff
@@ -0,0 +1,436 @@
+import { PageBlock } from 'notion-types'
+import { getPagePropertyFromId } from 'notion-utils'
+import * as React from 'react'
+import { useNotionContext } from '../context'
+import { DefaultPageIcon } from '../icons/default-page-icon'
+import SvgLeftChevron from '../icons/left-chevron'
+import SvgRightChevron from '../icons/right-chevron'
+import { CollectionViewProps } from '../types'
+import { cs, getWeeksInMonth } from '../utils'
+import { CollectionGroup } from './collection-group'
+import { getCollectionGroups } from './collection-utils'
+import { Property } from './property'
+
+const defaultBlockIds = []
+const currentYear = new Date(Date.now())
+const months = [
+  'January',
+  'February',
+  'March',
+  'April',
+  'May',
+  'June',
+  'July',
+  'August',
+  'September',
+  'October',
+  'November',
+  'December'
+]
+
+const monthsShort = [
+  'Jan',
+  'Feb',
+  'Mar',
+  'Apr',
+  'May',
+  'Jun',
+  'Jul',
+  'Aug',
+  'Sep',
+  'Oct',
+  'Nov',
+  'Dec'
+]
+let currentMonth = 0
+
+export const CollectionViewCalendar: React.FC<CollectionViewProps> = ({
+  collection,
+  collectionView,
+  collectionData
+}) => {
+  const isGroupedCollection = collectionView?.format?.collection_group_by
+
+  if (isGroupedCollection) {
+    const collectionGroups = getCollectionGroups(
+      collection,
+      collectionView,
+      collectionData
+    )
+
+    return collectionGroups.map((group, index) => (
+      <CollectionGroup
+        key={index}
+        {...group}
+        collectionViewComponent={Calendar}
+      />
+    ))
+  }
+
+  const blockIds =
+    (collectionData['collection_group_results']?.blockIds ??
+      collectionData['results:relation:uncategorized']?.blockIds ??
+      collectionData.blockIds) ||
+    defaultBlockIds
+
+  return (
+    <Calendar
+      collectionView={collectionView}
+      collection={collection}
+      blockIds={blockIds}
+    />
+  )
+}
+
+function Calendar({ blockIds, collectionView, collection }) {
+  const {
+    showCalendarControls,
+    startWeekOnMonday,
+    components,
+    recordMap,
+    mapPageUrl
+  } = useNotionContext()
+
+  const [weeksArr, setWeeksArr] = React.useState(
+    getWeeksInMonth(
+      currentYear.getFullYear(),
+      currentYear.getMonth(),
+      startWeekOnMonday
+    )
+  )
+
+  // Check if collection view is a standalone page or not
+  const isCollectionViewPage =
+    recordMap.block[Object.keys(recordMap.block)[0]].value.type ==
+    'collection_view_page'
+
+  const nextMonth = () => {
+    if (currentYear.getMonth() == 11) {
+      currentYear.setFullYear(currentYear.getFullYear() + 1)
+      currentYear.setMonth(0)
+    } else {
+      currentYear.setMonth(currentYear.getMonth() + 1)
+    }
+
+    currentMonth = 0
```

Comment:
Generally using globals like this in react is a bad practice except in really specific cases. in this case, I would recommend using `useState` or `useRef`. Keep in mind that it's possible for multiple calendar views to be visible on a page at a time (and that's generally the same constraint that all React components should adhere to).